src/cata_tiles.cpp

```diff
@@ -454,153 +741,182 @@ void cata_tiles::load_tilejson_from_file(JsonObject &config, int offset, int siz
         JsonObject entry = tiles.next_object();
 
         std::string t_id = entry.get_string("id");
-        tile_type *curr_tile = load_tile(entry, t_id, offset, size);
-        bool t_multi = entry.get_bool("multitile", false);
-        bool t_rota = entry.get_bool("rotates", t_multi);
+        tile_type &curr_tile = load_tile(entry, t_id, offset, size);
+
+        bool const t_multi = entry.get_bool("multitile", false);
+        bool const t_rota =  entry.get_bool("rotates", t_multi);
         if (t_multi) {
             // fetch additional tiles
             JsonArray subentries = entry.get_array("additional_tiles");
             while (subentries.has_more()) {
                 JsonObject subentry = subentries.next_object();
-                const std::string s_id = subentry.get_string("id");
-                const std::string m_id = t_id + "_" + s_id;
-                tile_type *curr_subtile = load_tile(subentry, m_id, offset, size);
-                curr_subtile->rotates = true;
-                curr_tile->available_subtiles.push_back(s_id);
+                std::string s_id = subentry.get_string("id");
+                std::string m_id = t_id + "_" + s_id;
+                
+                tile_type &curr_subtile = load_tile(subentry, m_id, offset, size);
+                curr_subtile.rotates = true;
+
+                multitile_variations_[t_id][from_string(s_id)] = &curr_subtile;
             }
         }
 
         // write the information of the base tile to curr_tile
-        curr_tile->multitile = t_multi;
-        curr_tile->rotates = t_rota;
+        curr_tile.multitile = t_multi;
+        curr_tile.rotates   = t_rota;
     }
-    dbg( D_INFO ) << "Tile Width: " << tile_width << " Tile Height: " << tile_height << " Tile Definitions: " << tile_ids.size();
+    dbg( D_INFO ) << "Tile Width: " << tile_width_ << " Tile Height: " << tile_height_ << " Tile Definitions: " << tile_ids.size();
 }
 
-tile_type *cata_tiles::load_tile(JsonObject &entry, const std::string &id, int offset, int size)
+tile_type& cata_tiles::load_tile(JsonObject &entry, const std::string &id,
+    int const offset, int const size)
 {
-    int fg = entry.get_int("fg", -1);
-    int bg = entry.get_int("bg", -1);
-    if (fg == -1) {
-        // OK, keep this value, indicates "doesn't have a foreground"
-    } else if (fg < 0 || fg >= size) {
+    static std::string const key_fg {"fg"}; 
+    static std::string const key_bg {"bg"};
+
+    // -1 indicates an absent fore / background.
+    int fg = entry.get_int(key_fg, -1);
+    int bg = entry.get_int(key_bg, -1);
+    if (fg != -1 && fg < 0 || fg >= size) {
```

Comment:
This is definitely wrong, it will decrement offset, which should almost certainly not happen.